### ISSUE TITLE
Non working Output-Items (f202.params)

### ISSUE THREAD

**Issue by suweg** [nbbrd/jdemetra-ws#19]
Hi,

we are having difficulties to configure our output items for the webservice, using the .params file. While most items are working, we do not get results for the following series items:

- sa 
- sa_f
- t
- t_f
- i
- i_f

We added "preprocessing" to other series items like "y" and "cal" and it worked for them. It does not work for the above mentioned series though. We also tried omitting the "preprocessing" and other variations, without success.

That's what our f202.params-file looks like:

```xml
<?xml version="1.0" encoding="UTF-8" standalone="true"?>
<wsaConfig ndecs="6" csvseparator="," csvlayout="list" bundle="10000">
    <policy>none</policy>
    <tsmatrix>
        <series>preprocessing.sa</series>
        <series>preprocessing.sa_f</series>
        <series>preprocessing.t</series>
        <series>preprocessing.t_f</series>
        <series>preprocessing.i</series>
        <series>preprocessing.i_f</series>
    </tsmatrix>
</wsaConfig>
```

This is the output filter definition in the x13requests.xml generated by the jwsrequests tool:
```xml
<x13:OutputFilter>preprocessing.sa preprocessing.sa_f preprocessing.t preprocessing.t_f preprocessing.i preprocessing.i_f</x13:OutputFilter>
```

Is there a list of all possible output items (series and matrix) and their complete names for the params file? This would help us a lot!

Thanks and best regards,
Susanne

**Comment by palatej** [comment]
I put below the file with all the possible outputs generated by the cruncher. The WEB service should use the same options. Most of them are self explaining.
Following that list, sa, sa_f ... should work without any prefix, but I need to check in more detail. 

[all.txt](https://github.com/nbbrd/jdemetra-ws/files/4224206/all.txt)

**Comment by suweg** [comment]
Thanks for your reply and the list. I guess the web service does not use the same options. 

For example: 
See the params-file attached. This is the output-filter in the x13-requests.xml, generated by the JWSRequests tool. 

	<x13:OutputFilter>preprocessing.y preprocessing.y_f sa sa_f t t_f i i_f</x13:OutputFilter>

A screenshot of the output is also attached.

- The web service output gives the y and y_t series although, according to your list, there should not be a "preprocessing.". I am getting no output of the y and y_t series, if I omit the "preprocessing."
- There is no output for the series " sa sa_f t t_f i i_f " at all. I tried several prefixes, without success. It does not work without a prefix either.

Regards
Susanne

-----Ursprüngliche Nachricht-----
Von: Jean Palate [mailto:notifications@github.com] 
Gesendet: Mittwoch, 19. Februar 2020 10:39
An: nbbrd/jdemetra-ws
Cc: Wegner, Susanne (C303); Author
Betreff: [EXTERN]Re: [nbbrd/jdemetra-ws] Non working Output-Items (f202.params) (#19)

I put below the file with all the possible outputs generated by the cruncher. The WEB service should use the same options. Most of them are self explaining.
Following that list, sa, sa_f ... should work without any prefix, but I need to check in more detail.

all.txt <https://github.com/nbbrd/jdemetra-ws/files/4224206/all.txt>

**Comment by suweg** [comment]
Hi, 

I need to reopen this issue, because we are resuming our own development work. 

We temporarily fixed the problem above by adding the prefix "final" to the desired output items. But the problem remains, that the names of the output items in the configuration file for the webservice and the crucher are different. 

Some background information about why we are rolling this issue up: We are about to switch from the webservice to the cruncher. But we still need the XML-output from the webservice, because we transform the data automatically into other formats, which does not work very well with the CSV-files, especially the matrix elements. So we are currently using the XML-generating classes from the Webservice together with the Cruncher. I am going to make a separate feature request for the cruncher for that.

Regards
Susanne

**Comment by suweg** [comment]
Sorry, I erroneously closed the issue.

**Comment by palatej** [comment]
Hi Suzanne, 

I found some time to check the code. In fact, you can use in the cruncher the same identifiers as in the WEB service (not the opposite). So, if you want to get the final sa, identified by "final.sa", you can put in the params file of the cruncher:

<series>final.sa</series>

That will generate the file "series_final_sa.csv"

The actual implementation is what you see in the WEB service. The simplified identifiers of the cruncher and of the graphical interface are workarounds to get simpler outputs and to be compatible with older releases.

Jean

**Comment by suweg** [comment]
Hi Jean,

thanks for checking and for your explanation. I am fine with whatever identifiers are necessary. But it would be nice for us to have a list of the actually implemented identifiers. 

In the reference manual, only the short versions are listed. Another important reason for this request is, that we are generating the params file semi-automatically from user input. The users, however, can only put in, what they find in the official documentation. Having a list, I could translate them in the background (just as you are doing it in your workaround).

Susanne

**Comment by palatej** [comment]
You are right. I just need to find time to do it. Probably during this summer. I will put it in the Wiki of this project

**Comment by suweg** [comment]
Thank you!
